docs/query-help-style-guide.md

```diff
@@ -0,0 +1,202 @@
+
+1.  `overview`—a summary of the purpose of the query, including an explanation of what the code does and the how the behavior of the program is affected.
+2.  `recommendation`—information on how to fix the issue highlighted by the query.
+3.  `example`—an example of code showing the problem. Where possible, this section should also include a solution to the issue.
+4.  `references`—relevant references to best practice. 
+
+For further information about the other section-level, block, list and table elements supported by the qhelp format, see [Qhelp files](https://wiki.semmle.com/display/SD/Qhelp+files).
+
+
+## English style
+
+You should write the overview and recommendation elements in simple English that is easy to follow. You should:
+
+*   Use simple sentence structures and avoid complex or academic language.
+*   Avoid colloquialisms and contractions.
+*   Use US English spelling throughout.
+*   Use words that are in common usage.
+
+## Code examples
+
+Whenever possible, you should include a code example that helps to explain the issue you are highlighting. Any code examples that you include should adhere to the following guidelines:
+
+*   The example should be less than 20 lines and ideally it should be runnable. If this is not possible, ensure that the example very clearly illustrates the issue that you are highlighting.
+*   Put the code example after the recommendation element where possible. Only include an example in the description element if absolutely necessary.
+*   If you are using an example to illustrate the solution to a problem, and the change required is minor, avoid repeating the whole example. It is preferable to either describe the change required or to include a smaller snippet of the corrected code.
+*   Clearly indicate which of the samples is an example of bad coding practice and which is recommended practice.
+
+There are several ways of including a code example in a query help file:
+
+*   Directly include the code as nested text in the `example` section, Use the `sample` block and define a `language` attribute to specify the language of the example:
+
+```       
+<example>
+<p>This example highlights poor coding practice</p>
+<sample language = "java" />
+
+Example of poor code
+
+<p>This example shows how to fix the code</p>
+<sample language = "java" />
+
+Example of correct code
+
+</example>
+```     
+
+
+*   Define the code example in a `src` file. The language is inferred from the file extension:
+
+```
+<example>
+<p>This example highlights poor coding practice</p>
+
+<sample src = "example-code-bad.java" />
+
+<p>This example shows how to fix the code</p>
+
+<sample src = "example-code-fixed.java" />
+</example>
+```
+     
+## Including references
+
+You should include one or more references to provide further information about the problem that your query is designed to find. References can be of the following types:
```

Comment:
It should probably be stated that the referenced should be a list formatted with `<li> .. </li>` for each item.